### PR TITLE
feat: add support for Kubernetes 1.15.0-rc.1 

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:
See https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.15.md#changelog-since-v1150-beta2

**Issue Fixed**:

**Requirements**:
- [x] Windows artifacts uploaded to acsmirror blob store
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
I didn't revert CoreDNS to 1.3.1 as the decision to do so was largely driven by configuration migration and "merged too close to release" issues that I don't think AKS Engine shares, but see the discussion at kubernetes/kubernetes#78691.

### CODE REVIEW

File: pkg/api/k8s_versions.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ var k8sComponentVersions = map[string]map[string]string{
 		"aci-connector":                    "virtual-kubelet:latest",
 		ContainerMonitoringAddonName:       "oms:ciprod01092019",
 		AzureCNINetworkMonitoringAddonName: "networkmonitor:v0.0.6",
-		"cluster-autoscaler":               "cluster-autoscaler:v1.14.2",
+		"cluster-autoscaler":               "cluster-autoscaler:v1.15.0",
```
Comment (jackfrancis) [inline pkg/api/k8s_versions.go]: Fun!

**Comment by mboersma** [comment]
codecov seems to have failed us, so I'm merging manually.

**Pull request merged**